### PR TITLE
Redesign command line interface

### PR DESCRIPTION
**Pull request by lucc**
This is the corresponding PR for #179.  Until now the unification of show-export and edit-source is done. The unification of the different ls versions is underway.

I am happy about any feedback.

### CODE REVIEW

**Comment by lucc** [comment]
There is one problem with the unification of the different ls subcommands. At least the phone subcommand tries to match the search terms against the phone numbers and not the rest of the card.  This makes sense but is harder to unify with the other listing like subcommands. I think I will try to rework the matching of cards first to have a better documented api for matching cards in specific ways. This could also help with implementing something like #131.

In conclusion I will leave this PR as is and only add more code in a new PR.

**Comment by lucc** [comment]
@scheibler as this feels like a fundamental decision to me and is also user facing I would like to have a short ok/no from you before I proceed.

**Comment by lucc** [comment]
@scheibler  ping?

**Comment by legrostdg** [comment]
@scheibler ping?
